internal/licenses/scanner.go

```diff
@@ -8,33 +8,37 @@ import (
 	"github.com/google/licensecheck"
 
 	"github.com/anchore/syft/internal/log"
-	"github.com/anchore/syft/syft/file"
-	"github.com/anchore/syft/syft/pkg"
 )
 
 const (
-	DefaultCoverageThreshold     = 75 // determined by experimentation
-	DefaultIncludeLicenseContent = false
+	UnknownLicensePrefix     = unknownLicenseType + "_"
+	DefaultCoverageThreshold = 75 // determined by experimentation
+
+	unknownLicenseType = "UNKNOWN"
 )
 
+type Evidence struct {
+	ID    string            // License identifier. (See licenses/README.md.)
+	Type  licensecheck.Type // The type of the license: BSD, MIT, etc.
+	Start int               // Start offset of match in text; match is at text[Start:End].
+	End   int               // End offset of match in text.
+	IsURL bool              // Whether match is a URL.
+}
+
 type Scanner interface {
-	IdentifyLicenseIDs(context.Context, io.Reader) ([]string, []byte, error)
-	FileSearch(context.Context, file.LocationReadCloser) ([]file.License, error)
-	PkgSearch(context.Context, file.LocationReadCloser) ([]pkg.License, error)
+	FindEvidence(context.Context, io.Reader) ([]Evidence, []byte, error)
```

Comment:
Why return the full contents?
```suggestion
	FindEvidence(context.Context, io.Reader) ([]Evidence, error)
```

This way you could change evidence to:
```
type Evidence struct {
	ID    string            // License identifier. (See licenses/README.md.)
	Type  licensecheck.Type // The type of the license: BSD, MIT, etc.
	Start int               // Start offset of match in text; match is at text[Start:End].
	End   int               // End offset of match in text.
    Contents []byte
	IsURL bool              // Whether match is a URL.
}
```

Worth asking and chatting about it before refactoring with this suggestion.

Reply:
from an offline chat: there is a need when there are no Evidence objects returned to reference the entire contents (as a sha256)